server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1635,6 +1645,17 @@ private void expungeVolumesInPrimaryOrSecondary(VolumeVO volume, DataStoreRole r
             }
         }
     }
+
+    private boolean canAccessVolumeStore(VolumeVO volume, DataStoreRole role) {
+        if (volume == null) {
+            throw new CloudRuntimeException("Cannot find the volume");
```

Comment:
this is a strange message in this place. We are not trying to find it here, but we claim that is what failed. maybe this check should be somewhere else or the message should be adjusted to "No Volume given" or so.

Reply:
Thanks @DaanHoogland, fixed